src/main/flight/imu.c

```diff
@@ -342,3 +342,13 @@ int16_t calculateThrottleAngleCorrection(uint8_t throttle_correction_value)
         angle = 900;
     return lrintf(throttle_correction_value * sinf(angle / (900.0f * M_PIf / 2.0f)));
 }
+
+int16_t calculateThrottleAngleCorrectionAxis(uint8_t throttle_hovering_percent, uint8_t axis){
+
+    if (ABS(anglerad[axis]) > M_PIf/2)
+        return 0; //if upside down don't increase correction
+
+    uint16_t liftOffTrust = throttle_hovering_percent * 10; //to bring from 0-100 range to 0-1000 range (thrust input is in the 1000-2000 range)
+
+    return lrintf( constrainf(  liftOffTrust / cosf( ABS(anglerad[axis]) ) - liftOffTrust, 0, 1000 ) );
```

Comment:
That's right. If you tilt the multirotor while climbing, it will continue climbing. If you want to stop climbing, pull the throttle down (: That would be correct and expected behavior.